pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -489,6 +490,14 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *api.Pod, r
 						break
 					}
 				}
+				if containerSpec == nil {
+					for i, c := range pod.Spec.InitContainers {
```

Comment:
Since this for loop is duplicated verbatim, maybe it's worth having a `func (containerlist) getContainerByName(name) *container` helper. Fine either way, just saves some indentation.